diffkemp/simpll/ModuleComparator.cpp

```diff
@@ -95,6 +95,24 @@ void ModuleComparator::compareFunctions(Function *FirstFun,
 
             ConstFunPair missingDefs;
             bool inlined = false;
+            // If inlining two functions with the same name, it is possible that
+            // there is a difference in their bodies and only a formal
+            // difference that won't show up in the syntax diff.
+            // For this reason, the functions are compared using
+            // ModuleComparator and the original functions are marked as
+            // covered, leading to the diff not being show when empty.
+            Function *InlinedFunFirst = !inlineFirst ? nullptr :
+                    getCalledFunction(inlineFirst->getCalledValue());
+            Function *InlinedFunSecond = !inlineSecond ? nullptr :
+                    getCalledFunction(inlineSecond->getCalledValue());
+            if (InlinedFunFirst && InlinedFunSecond &&
+                !InlinedFunFirst->getName().startswith("simpll__") &&
```

Comment:
This can replaced by `isSimpllAbstractionDeclaration`.

Reply:
That wouldn't work, because I also have to exclude field access abstractions - `isSimpllAbstractionDeclaration` checks only for abstraction declarations (i.e. those generated by `FunctionAbstractionGenerator`).

Reply:
Then let's write a function for checking field access abstractions?